client/src/ui/js/custom.js

```diff
@@ -0,0 +1,159 @@
+        clearInterval(timer);
+        const resBtn = document.getElementById("res-btn");
+        resBtn.disabled = false;
+        resBtn.classList.remove("btn-secondary");
+        resBtn.classList.add("btn-success");
+        query = '#res-btn';
+        buttonClick(query, SAML_RESPONSE_TITLE, res, res, VALUE_TYPE);
+        blinking(query)
+    }
+}
+
+function addOIDCConfig(oidcAuthzRequest, oidcAuthzResponse, oidcTokenRequest, oidcTokenResponse) {
+
+    const OIDC_AUTHZ_REQUEST = "OIDC_AUTHZ_REQUEST";
+    const OIDC_AUTHZ_REQUEST_TITLE = "OIDC Authorization Request";
+    const OIDC_AUTHZ_RESPONSE = "OIDC_AUTHZ_RESPONSE";
+    const OIDC_AUTHZ_RESPONSE_TITLE = "OIDC Authorization Response";
+    const OIDC_TOKEN_REQUEST = "OIDC_TOKEN_REQUEST";
+    const OIDC_TOKEN_REQUEST_TITLE = "OIDC Token Request";
+    const OIDC_TOKEN_RESPONSE = "OIDC_TOKEN_RESPONSE";
+    const OIDC_TOKEN_RESPONSE_TITLE = "OIDC Token Response";
+    const VALUE_TYPE = "json";
+
+    let curBtn;
+    let query;
+    if (!oidcAuthzRequest.includes(OIDC_AUTHZ_REQUEST)) {
+        let card = document.getElementById("oidc-label");
+        card.classList.add("alert-primary");
+        curBtn = document.getElementById("authz-req-btn");
+        curBtn.disabled = false;
+        curBtn.classList.remove("btn-secondary");
+        curBtn.classList.add("btn-primary");
+        query = '#authz-req-btn';
+        buttonClick(query, OIDC_AUTHZ_REQUEST_TITLE, oidcAuthzRequest, null, VALUE_TYPE);
+        blinking(query);
+    }
+    if (!oidcAuthzResponse.includes(OIDC_AUTHZ_RESPONSE)) {
+        clearInterval(timer);
+        curBtn = document.getElementById("authz-res-btn");
+        curBtn.disabled = false;
+        curBtn.classList.remove("btn-secondary");
+        curBtn.classList.add("btn-success");
+        query = '#authz-res-btn';
+        buttonClick(query, OIDC_AUTHZ_RESPONSE_TITLE, oidcAuthzRequest, oidcAuthzResponse, VALUE_TYPE);
+        blinking(query)
+    }
+    if (!oidcTokenRequest.includes(OIDC_TOKEN_REQUEST)) {
+        clearInterval(timer);
+        curBtn = document.getElementById("token-req-btn");
+        curBtn.disabled = false;
+        curBtn.classList.remove("btn-secondary");
+        curBtn.classList.add("btn-primary");
+        query = '#token-req-btn';
+        buttonClick(query, OIDC_TOKEN_REQUEST_TITLE, oidcTokenRequest, null, VALUE_TYPE);
+        blinking(query)
+    }
+    if (!oidcTokenResponse.includes(OIDC_TOKEN_RESPONSE)) {
+        clearInterval(timer);
+        curBtn = document.getElementById("token-res-btn");
+        curBtn.disabled = false;
+        curBtn.classList.remove("btn-secondary");
+        curBtn.classList.add("btn-success");
+        query = '#token-res-btn';
+        oidcTokenResponse = oidcTokenResponse.replace(/"\{/g, '{');
+        oidcTokenResponse = oidcTokenResponse.replace(/\}"/g, '}');
+        buttonClick(query, OIDC_TOKEN_RESPONSE_TITLE, oidcTokenRequest, oidcTokenResponse, VALUE_TYPE);
```

Comment:
Add doc comments explaining each function and their parameters for better understandability of the code.

Reply:
Fixed